client/mm/mm_arb_market_maker.go

```diff
@@ -82,6 +82,57 @@ type ArbMarketMakerConfig struct {
 	NumEpochsLeaveOpen uint64                      `json:"orderPersistence"`
 }
 
+func (a *ArbMarketMakerConfig) copy() *ArbMarketMakerConfig {
+	c := *a
+
+	c.BuyPlacements = make([]*ArbMarketMakingPlacement, 0, len(a.BuyPlacements))
+	for _, p := range a.BuyPlacements {
+		c.BuyPlacements = append(c.BuyPlacements, &ArbMarketMakingPlacement{
+			Lots:       p.Lots,
+			Multiplier: p.Multiplier,
+		})
+	}
+
+	c.SellPlacements = make([]*ArbMarketMakingPlacement, 0, len(a.SellPlacements))
+	for _, p := range a.SellPlacements {
+		c.SellPlacements = append(c.SellPlacements, &ArbMarketMakingPlacement{
+			Lots:       p.Lots,
+			Multiplier: p.Multiplier,
+		})
+	}
+
+	return &c
+}
+
+// updateLotSize modifies the number of lots in each placement in the event
+// of a lot size change. It will place as many lots as possible without
+// exceeding the total quantity placed using the original lot size.
+//
+// This function is NOT thread safe.
+func (c *ArbMarketMakerConfig) updateLotSize(originalLotSize, newLotSize uint64) {
+	for _, p := range c.SellPlacements {
+		p.Lots = (p.Lots * originalLotSize) / newLotSize
+	}
+
+	for _, p := range c.BuyPlacements {
+		p.Lots = (p.Lots * originalLotSize) / newLotSize
+	}
+}
```

Comment:
Let's say we have 4 sell placements of 1 lot each. If we double the lot size, this will result in 4 placements of 0 lots. Instead, I think we should focus on preserving total quantity and prioritizing higher-priority placements, so that we end up with 2 placements of 1 lot instead.

```Go
func (c *ArbMarketMakerConfig) updateLotSize(originalLotSize, newLotSize uint64) error {
	updatedPlacements := func(oldPlacements []*ArbMarketMakingPlacement) (newPlacements []*ArbMarketMakingPlacement) {
		var qtyCounter uint64
		for _, p := range oldPlacements {
			qtyCounter += p.Lots * originalLotSize
		}
		newPlacements = make([]*ArbMarketMakingPlacement, 0, len(oldPlacements))
		for _, p := range oldPlacements {
			lots := utils.Max((p.Lots*originalLotSize)/newLotSize, 1) // might want to round instead of truncate
			qty := lots * newLotSize
			if qty > qtyCounter {
				return
			}
			qtyCounter -= qty
			newPlacements = append(newPlacements, &ArbMarketMakingPlacement{
				Lots:      lots,
				Multiplier: p.Multiplier,
			})
		}
		return
	}

	if len(c.SellPlacements) > 0 {
		c.SellPlacements = updatedPlacements(c.SellPlacements)
		if len(c.SellPlacements) == 0 {
			return errors.New("no sell placements possible with new lot size")
		}
	}
	if len(c.BuyPlacements) > 0 {
		c.BuyPlacements = updatedPlacements(c.BuyPlacements)
		if len(c.BuyPlacements) == 0 {
			return errors.New("no buy placements possible with new lot size")
		}
	}
	return nil
}
```

Either way, we need to validate the resulting placements to ensure there are none with zero lots, especially since `ArbMarketMakerConfig` doesn't have a `Validate` method.